scripts/onboarding/aks/onboarding-msi-bicep/existingClusterOnboarding.bicep

```diff
@@ -0,0 +1,111 @@
+param resourceTagValues object
+
+@description('Workspace Region for data collection rule')
+param workspaceRegion string
+
+@description('Full Resource ID of the log analitycs workspace that will be used for data destination. For example /subscriptions/00000000-0000-0000-0000-0000-00000000/resourceGroups/ResourceGroupName/providers/Microsoft.operationalinsights/workspaces/ws_xyz')
+param workspaceResourceId string
+
+@description('Data collection interval e.g. "5m" for metrics and inventory. Supported value range from 1m to 30m')
+param dataCollectionInterval string
+
+@description('Data collection Filtering Mode for the namespaces')
+@allowed([
+  'Off'
+  'Include'
+  'Exclude'
+])
+param namespaceFilteringModeForDataCollection string = 'Off'
+
+@description('An array of Kubernetes namespaces for the data collection of inventory, events and metrics')
+param namespacesForDataCollection array
+
+var clusterSubscriptionId = split(aksResourceId, '/')[2]
+var clusterResourceGroup = split(aksResourceId, '/')[4]
+var clusterName = split(aksResourceId, '/')[8]
+var workspaceLocation = replace(workspaceRegion, ' ', '')
+var dcrNameFull = 'MSCI-${workspaceLocation}-${clusterName}'
+var dcrName = ((length(dcrNameFull) > 64) ? substring(dcrNameFull, 0, 64) : dcrNameFull)
+var associationName = 'ContainerInsightsExtension'
+var dataCollectionRuleId = resourceId(clusterSubscriptionId, clusterResourceGroup, 'Microsoft.Insights/dataCollectionRules', dcrName)
+
+resource aks_monitoring_msi_dcr 'Microsoft.Insights/dataCollectionRules@2022-06-01' = {
+  name: dcrName
+  location: workspaceRegion
+  tags: resourceTagValues
+  kind: 'Linux'
+  properties: {
+    dataSources: {
+      extensions: [
+        {
+          name: 'ContainerInsightsExtension'
+          streams: [
+            'Microsoft-ContainerInsights-Group-Default'
+          ]
+          extensionSettings: {
+            dataCollectionSettings: {
+              interval: dataCollectionInterval
+              namespaceFilteringMode: namespaceFilteringModeForDataCollection
+              namespaces: namespacesForDataCollection
+            }
+          }
+          extensionName: 'ContainerInsights'
+        }
+      ]
+    }
+    destinations: {
+      logAnalytics: [
+        {
+          workspaceResourceId: workspaceResourceId
+          name: 'ciworkspace'
+        }
+      ]
+    }
+    dataFlows: [
+      {
+        streams: [
+          'Microsoft-ContainerInsights-Group-Default'
+        ]
+        destinations: [
+          'ciworkspace'
+        ]
+      }
+    ]
+  }
+}
+
+resource aks_monitoring_msi_addon 'Microsoft.ContainerService/managedClusters@2018-03-31' = {
+  name: clusterName
+  location: aksResourceLocation
+  tags: resourceTagValues
+  properties: {
+    addonProfiles: {
+      omsagent: {
+        enabled: true
+        config: {
+          logAnalyticsWorkspaceResourceID: workspaceResourceId
+          useAADAuth: 'true'
+        }
+      }
+    }
+  }
```

Comment:
aks_monitoring_msi_dcra should be added as dependency for aks_monitoring_msi_addon